### DISCUSSION TITLE
Add JSON representation for valueString

### DISCUSSION THREAD

**Discussion by kurzroman** [grafana/grafana#45457] [Alerting]
Hi all,

The json body of the webhook contact point in the new ng alert features contains `valueString` and looks like this:
`[ metric='cheetah ' labels={host=cheetah} value=1 ], [ metric='flagbenny ' labels={host=flagbenny} value=1 ], [ metric='insect ' labels={host=insect} value=1 ], [ metric='mora ' labels={host=mora} value=1 ], [ metric='mummichog ' labels={host=mummichog} value=1 ], [ metric='razorfish ' labels={host=razorfish} value=5 ], [ metric='ronquil ' labels={host=ronquil} value=1 ]
`

I would suggest that the webhook body already contains some sort of JSON representation for `valueString`.

In my opinion it's really hard to work with the current `valueString` in external applications. Right now we need custom code and some sort of scatchy regex to parse this and work with it.

```js
function parseValueString(valueString) {
    let resultValues = [];
    for (let match of valueString.matchAll(/\[\s*(var|metric)='([^']+)'\s+labels={([^}]*)}\s+value=([^\s]+)\s*\]/g)) {
        let value = {
            type: match[1],
            metric: match[2],
            labels: {},
            value: parseFloat(match[4])
        };
        if (match[3].length > 0) {
            for (let labelPair of match[3].split(",")) {
                let labelParts = labelPair.split("=");
                if (labelParts.length === 2) {
                    value.labels[labelParts[0].trim()] = labelParts[1].trim();
                }
            }
        }
        resultValues.push(value);
    }
    return resultValues;
}
```

**Comment by Fuco1** [comment]
We are trying to use the webhooks to format alerts for slack in a nicer way (i.e. buttons for links instead of unreadable URLs).  The value string as json would help a lot making the whole thing easier.

**Comment by gtirloni** [comment]
I'm seriously frustrated at Grafana for ignoring this for so long. The metric's labels are buried in ValueString and there doesn't seem to have any other way to get to them besides using regular expressions. This is a clear regression from the legacy alerting system.

**Comment by mellieA** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

As we were reviewing discussions to migrate them, we noticed that you're asking about an alerting feature that we've improved our docs around. 

I'd suggest trying to format the webhook alert by using the go template format to create an annotation that includes the labels and values from the alert response instead of sending just the valueString. 

Check out the docs we've created, this was initially not a well-documented process but we've been working on it: 

https://grafana.com/blog/2023/04/05/grafana-alerting-a-beginners-guide-to-templating-alert-notifications/
https://grafana.com/docs/grafana/latest/alerting/manage-notifications/template-notifications/using-go-templating-language/
https://grafana.com/docs/grafana/latest/alerting/manage-notifications/template-notifications/reference/#reference 

If you still need help, please consider asking the community for help in our forum at [community.grafana.com](https://community.grafana.com/c/support/alerting/60) or open a new feature request as an issue.
